commit 3bb547c882077b74e4e3c009eb9c9cf734ee7ce9 (refs/changes/38/5238/3)
Author: Josh Wilsdon <josh@wilsdon.ca>
Date:   2018-12-13T23:17:58-08:00 (10 months ago)
    
    TRITON-1036 vmapi should reject invalid image uuids for reprovision
    Reviewed by: Trent Mick <trentm@gmail.com>
    Approved by: Trent Mick <trentm@gmail.com>

diff --git a/lib/endpoints/vms.js b/lib/endpoints/vms.js
index 529b274..c239556 100644
--- a/lib/endpoints/vms.js
+++ b/lib/endpoints/vms.js
@@ -659,6 +659,7 @@ function rebootVm(req, res, next) {
  */
 function reprovisionVm(req, res, next) {
     req.log.trace({ vm_uuid: req.params.uuid }, 'ReprovisionVm start');
+    var error;
     var vm = req.vm;
 
     if (['joyent', 'joyent-minimal', 'lx'].indexOf(vm.brand) === -1) {
@@ -667,7 +668,13 @@ function reprovisionVm(req, res, next) {
     }
 
     if (!req.params.image_uuid) {
-        var error = [ errors.missingParamErr('image_uuid') ];
+        error = [ errors.missingParamErr('image_uuid') ];
+        return next(new errors.ValidationFailedError('Invalid Parameters',
+            error));
+    }
+
+    if (!common.validUUID(req.params.image_uuid)) {
+        error = [ errors.invalidUuidErr('image_uuid') ];
         return next(new errors.ValidationFailedError('Invalid Parameters',
             error));
     }
diff --git a/package.json b/package.json
index 95e3989..a4df7f3 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "vmapi",
   "description": "VMs API",
-  "version": "9.6.5",
+  "version": "9.6.6",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
